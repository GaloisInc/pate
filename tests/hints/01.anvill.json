{
    "symbols": [
        [
            4195440,
            "printf"
        ],
        [
            4195712,
            "main"
        ],
        [
            4195424,
            "puts"
        ],
        [
            400,
            "errno"
        ]
    ],
    "variables": [
        { "type" : "i",
          "address" : 400
        }
    ],
    "functions": [
        {
            "is_variadic": true,
            "return_values": [
                {
                    "type": "i",
                    "register": "RAX"
                }
            ],
            "parameters": [
                {
                    "type": "*b",
                    "register": "RDI"
                }
            ],
            "return_stack_pointer": {
                "type": "L",
                "offset": 8,
                "register": "RSP"
            },
            "return_address": {
                "type": "L",
                "memory": {
                    "offset": 0,
                    "register": "RSP"
                }
            },
            "address": 4195440
        },
        {
            "return_values": [
                {
                    "type": "l",
                    "register": "RAX"
                }
            ],
            "return_stack_pointer": {
                "type": "L",
                "offset": 8,
                "register": "RSP"
            },
            "return_address": {
                "type": "L",
                "memory": {
                    "offset": 0,
                    "register": "RSP"
                }
            },
            "address": 4195712
        },
        {
            "return_values": [
                {
                    "type": "i",
                    "register": "RAX"
                }
            ],
            "parameters": [
                {
                    "type": "i",
                    "register": "RDI"
                }
            ],
            "return_stack_pointer": {
                "type": "L",
                "offset": 8,
                "register": "RSP"
            },
            "return_address": {
                "type": "L",
                "memory": {
                    "offset": 0,
                    "register": "RSP"
                }
            },
            "address": 4195424
        }
    ],
    "os": "linux",
    "arch": "amd64"
}
