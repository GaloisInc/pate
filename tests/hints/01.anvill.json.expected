VerificationHints { blockMappings = []
                  , functionEntries = [ ("printf", 4195440)
                                      , ("main", 4195712)
                                      , ("puts", 4195424)
                                      ]
                  , dataSymbols = [ ("errno", SymbolExtent { symbolLocation = 400, symbolSize = 4} )
                                  ]
                  }
